pkg/controller/installation/windows_controller.go

```diff
@@ -366,7 +361,7 @@ func (r *ReconcileWindows) Reconcile(ctx context.Context, request reconcile.Requ
 		nodePrometheusTLS, err = certificateManager.GetKeyPair(r.client, render.NodePrometheusTLSServerSecret, common.OperatorNamespace(), dns.GetServiceDNSNames(render.CalicoNodeMetricsService, common.CalicoNamespace, r.clusterDomain))
 		if err != nil {
 			r.status.SetDegraded(operatorv1.ResourceCreateError, "Error getting TLS certificate", err, reqLogger)
-			return reconcile.Result{RequeueAfter: 10 * time.Second}, err
```

Comment:
Yeah I think this should be covered by adding a watch for that.

As for not having requeue and an error, I think as long as we have the Periodic reconciles we're probably good dropping the requeues. Because my concern would be when there is an issue reaching the K8s api that eventually errors would be dropped but with the periodic requeue everything would be eventually reconciled.

Reply:
Yeah, feels like the pattern we want to be establishing and moving towards is:

- Watch anything that you need to be notified of.
- Add a periodic reconcile as a backstop
- Never use RequeueAfter